src/js/iconpicker.js

```diff
@@ -741,24 +772,30 @@ module.exports = allIconNames;
                     return $(false);
                 } else {
                     var found = [];
+                    var show = [];
+                    var hide = [];
+
+                    var regex = false;
+                    try {
+                        regex = new RegExp('(?:^|\\W)' + filterText);
+                    } catch (e) {
+                        regex = false;
+                    }
+
                     this.iconpicker.find('.iconpicker-item').each(function() {
                         var $this = $(this);
                         var text = $this.attr('title').toLowerCase();
                         var searchTerms = $this.attr('data-search-terms') ? $this.attr('data-search-terms').toLowerCase() : '';
                         text = text + ' ' + searchTerms;
-                        var regex = false;
-                        try {
-                            regex = new RegExp('(^|\\W)' + filterText, 'g');
-                        } catch (e) {
-                            regex = false;
-                        }
-                        if ((regex !== false) && text.match(regex)) {
+                        if ((regex !== false) && regex.test(text)) {
                             found.push($this);
-                            $this.show();
+                            show.push(this);
```

Comment:
Should this be $this?
```suggestion
                            show.push($this);
```

Reply:
It's intended - must be a list of DOM elements to be supplied to [this jquery collection constructor](https://api.jquery.com/jQuery/#jQuery-elementArray) used below.

There doesn't seem to be a constructor with a list of other jquery collections, but admittedly I didn't check if it works in practice. I can make a quick test.